scipy/linalg/_special_matrices.py

```diff
@@ -1012,23 +1041,28 @@ def fiedler_companion(a):
 
     Parameters
     ----------
-    a : (N,) array_like
+    a : (..., N) array_like
         1-D array of polynomial coefficients in descending order with a nonzero
         leading coefficient. For ``N < 2``, an empty array is returned.
+        M-dimensional arrays are treated as a batch: each slice along the last
```

Comment:
Wasn't sure if this was intentional as elsewhere you use
```suggestion
        N-dimensional arrays are treated as a batch: each slice along the last
```

Reply:
Good catch. This is why I don't think we should try to customize the documentation like this for each function, at least not initially - it's hard to write in a way that's easy to read, too easy to get wrong, and time consuming for the author and reviewer. gh-21462 proposes generic documentation and a tutorial.

Reply:
Would you prefer if I review #21462 first?

Reply:
I think it's ok to start here, thanks. That way we can see how function-specific batch documentation looks, what it adds for the reader, and whether it complicates review. That will help inform decisions in gh-21462. (And actually, let's do gh-21419 between these two if you're willing because that would finish the special matrix functions.)